src/Button/Button.tsx

```diff
@@ -0,0 +1,225 @@
+import * as React from 'react';
+import cx from '../utils/classnames';
+import every = require('lodash/every');
+import { props, t, stateClassUtil } from '../utils';
+import _TextOverflow from '../TextOverflow/TextOverflow';
+import { PopoverProps } from '../Popover/Popover';
+import { ObjectOverwrite } from 'typelevel-ts';
+import FlexView from 'react-flexview';
+import Icon from '../Icon/Icon';
+import LoadingSpinner from '../LoadingSpinner/LoadingSpinner';
+
+export namespace ButtonProps {
+
+  export type ButtonState = 'ready' | 'not-allowed' | 'processing' | 'error' | 'success';
+  export type ButtonType = 'default' | 'primary' | 'positive' | 'negative' | 'flat';
+  export type ButtonSize = 'tiny' | 'small' | 'medium';
+
+}
+
+// TODO: TextOverflow should accept a popoverProps
+export type TextOverflowCompatibleComponent = React.ComponentClass<{
+  label?: string | number,
+  popover?: ObjectOverwrite<PopoverProps.Popover, {
+    position?: PopoverProps.Position,
+    content?: void & string
+  }>
+}>
+
+export type ButtonStateMap = { [key in ButtonProps.ButtonState]?: string };
+export interface ButtonRequiredProps {
+  /** callback */
+  onClick: (e: React.SyntheticEvent<HTMLDivElement>) => void
+  /** can be a String, or a dictionary { [buttonState]: String }, t.maybe(t.union([t.Str,  stringForButtonStates]) */
+  label?: string | ButtonStateMap
+  /** otherwise just pass a string as children */
+  children?: string
+  /** type of the button (default, primary, positive, negative, flat) */
+  type?: ButtonProps.ButtonType
+  /** shortcut for type "flat" */
+  flat?: boolean,
+}
+
+export interface ButtonDefaultProps {
+  /** function to handle the overflow of too long labels, replacing with ellipsed string and tooltip */
+  textOverflow: TextOverflowCompatibleComponent;
+  /** ready, not-allowed, processing, success, error; overrides `baseState`, use it if you want button to be a functional component */
+  buttonState: ButtonProps.ButtonState;
+  /** size of the button, one of 'tiny', 'small', 'medium' */
+  size: ButtonProps.ButtonSize;
+  /** fluid (block) button, takes the width of the container */
+  fluid: boolean;
+  /** shortcut for type "primary" */
+  primary: boolean;
+  /** circular button, this is allowed only if it's an icon button */
+  circular: boolean;
+  /** can be a String referring to an icon, or a dictionary { [buttonState]: String },t.maybe(t.union([t.Str, stringForButtonStates])) */
+  icon: string | ButtonStateMap;
+  /** an optional class name to pass to first inner element of the component */
+  className: string;
+  /** an optional style object to pass to first inner element of the component */
+  style: object
+};
+
+
+export type ButtonProps = ButtonRequiredProps & Partial<ButtonDefaultProps>;
+export type ButtonPropTypes = ButtonProps;
```

Comment:
What is this?

Reply:
It was exported in the `.d.ts` before, so I've just reported here. It was probably used in alinity-pro, but we can remove it, and then fix in the other projects (if needed)

Reply:
@gabro Should I delete it then?

Reply:
I would remove it, yes.

Reply:
Removed :)